```cpp
consteval bool ce_255() {
    big_int result{2};
    result *= 10;
    result += 5;
    result *= 10; // error here
    result += 5;
    return result == big_int{255};
}
static_assert(ce_255());
```

```
/__w/std-big-int/std-big-int/tests/beman/big_int/multiplication.test.cpp:48:15: error: static assertion expression is not an integral constant expression
   48 | static_assert(ce_255());
      |               ^~~~~~~~
/usr/lib/gcc/x86_64-pc-linux-gnu/15/include/g++-v15/bits/stl_algobase.h:585:18: note: assignment to member 'limbs' of union with active member 'data' is not allowed in a constant expression
  585 |               *__result = *__first;
      |                         ^
/usr/lib/gcc/x86_64-pc-linux-gnu/15/include/g++-v15/bits/stl_algo.h:698:20: note: in call to '__copy_n_a<const unsigned long long *, unsigned long, unsigned long long *>(&basic_big_int<64>{}.m_storage.limbs[0], 1, &result.m_storage.limbs[0], true)'
  698 |       auto __res = std::__copy_n_a(std::__niter_base(__first), __n2,
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  699 |                                    std::__niter_base(__result), true);
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/std-big-int/std-big-int/include/beman/big_int/big_int.hpp:458:13: note: in call to 'copy_n<const unsigned long long *, unsigned long, unsigned long long *>(&basic_big_int<64>{}.m_storage.limbs[0], 1, &result.m_storage.limbs[0])'
  458 |             std::copy_n(src_limbs, src_count, dst_limbs);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/std-big-int/std-big-int/include/beman/big_int/big_int.hpp:679:5: note: in call to 'this->assign_value<beman::big_int::basic_big_int<64>>(basic_big_int<64>{}, 0)'
  679 |     assign_value(std::move(x));
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/std-big-int/std-big-int/include/beman/big_int/big_int.hpp:2097:9: note: in call to '*this.operator=(basic_big_int<64>{})'
 2097 |         *this                    = basic_big_int{};
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/std-big-int/std-big-int/tests/beman/big_int/multiplication.test.cpp:44:5: note: in call to 'result.operator*=<int>(10)'
   44 |     result *= 10;
      |     ^~~~~~~~~~~~
/__w/std-big-int/std-big-int/tests/beman/big_int/multiplication.test.cpp:48:15: note: in call to 'ce_255()'
   48 | static_assert(ce_255());
      |               ^~~~~~~~
```

--- a/include/beman/big_int/big_int.hpp
+++ b/include/beman/big_int/big_int.hpp
@@ -663,13 +663,7 @@ constexpr basic_big_int<b, A>::basic_big_int(basic_big_int&& x) noexcept
         m_storage.data    = x.m_storage.data;
         x.m_capacity      = 0;
         x.m_size_and_sign = 1;
-        // x now looks like static storage (m_capacity == 0),
-        // but the active union member is still `data`.
-        // Reactivate `limbs`
-        // We can't use std::fill_n because it doesn't activate the member
-        for (size_type i = 0; i < inplace_capacity; ++i) {
-            x.m_storage.limbs[i] = limb_type{0};
-        }
+        x.m_storage = {};
     }
 }
 

--- a/include/beman/big_int/big_int.hpp
+++ b/include/beman/big_int/big_int.hpp
@@ -493,6 +493,9 @@ class BEMAN_BIG_INT_TRIVIAL_ABI basic_big_int {
                 m_storage.data      = src.m_storage.data;
                 src.m_capacity      = 0;
                 src.m_size_and_sign = 1;
+                for (size_type i = 0; i < inplace_capacity; ++i) {
+                    src.m_storage.limbs[i] = limb_type{0};
+                }
                 if (m_capacity < needed) {
                     grow(needed);
                 }
@@ -660,6 +663,13 @@ constexpr basic_big_int<b, A>::basic_big_int(basic_big_int&& x) noexcept
         m_storage.data    = x.m_storage.data;
         x.m_capacity      = 0;
         x.m_size_and_sign = 1;
+        // x now looks like static storage (m_capacity == 0),
+        // but the active union member is still `data`.
+        // Reactivate `limbs`
+        // We can't use std::fill_n because it doesn't activate the member
+        for (size_type i = 0; i < inplace_capacity; ++i) {
+            x.m_storage.limbs[i] = limb_type{0};
+        }
     }
 }
 

--- a/include/beman/big_int/big_int.hpp
+++ b/include/beman/big_int/big_int.hpp
@@ -493,9 +493,7 @@ class BEMAN_BIG_INT_TRIVIAL_ABI basic_big_int {
                 m_storage.data      = src.m_storage.data;
                 src.m_capacity      = 0;
                 src.m_size_and_sign = 1;
-                for (size_type i = 0; i < inplace_capacity; ++i) {
-                    src.m_storage.limbs[i] = limb_type{0};
-                }
+                src.m_storage.limbs[0] = limb_type{0};
                 if (m_capacity < needed) {
                     grow(needed);
                 }

